numcodecs/compat.py

```diff
@@ -14,29 +14,47 @@
 
 if PY2:  # pragma: py3 no cover
 
+    buffer = buffer
     text_type = unicode
     binary_type = str
     integer_types = (int, long)
     reduce = reduce
 
 else:  # pragma: py2 no cover
 
+    buffer = memoryview
     text_type = str
     binary_type = bytes
     integer_types = int,
     from functools import reduce
 
 
+def to_buffer(v):
+    """Obtain a `buffer` or `memoryview` for `v`."""
+    b = v
+    if isinstance(b, np.ndarray):
+        if b.dtype.kind is 'O':
+            raise ValueError('cannot encode object array')
+        b = b.reshape(-1, order='A')
+        if b.dtype.kind in 'Mm':
+            b = b.view(np.uint8)
```

Comment:
Minor point, but suggest to view this as int64, so itemsize is preserved.

Reply:
Thanks for catching this. Fixed.